app/models/compound_aliquot.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class CompoundAliquot # rubocop:todo Style/Documentation
+  include ActiveModel::Model
+
+  DUPLICATE_TAG_DEPTH_ERROR_MSG = "Cannot create compound sample from following samples due to duplicate 'tag depth'"
+  MULTIPLE_STUDIES_ERROR_MSG =
+    'Cannot create compound sample due to the component samples being under different studies.'
+
+  attr_accessor :request, :source_aliquots
+
+  attr_reader :component_samples
+
+  validate :tag_depth_is_unique
+
+  def initialize(attributes)
+    super
+
+    @component_samples = source_aliquots.map(&:sample)
+  end
+
+  # Check that the component samples in the compound sample will be able to be distinguished -
+  # this is represented by them all having a unique 'tag_depth'
+  def tag_depth_is_unique
+    return unless source_aliquots.pluck(:tag_depth).uniq.count != source_aliquots.size
```

Comment:
Should probably use size or count on both sides here. Given both are arrays I thing size, count and length are all essentially synonyms.

Also, you could do:

```
return unless source_aliquots.pluck(:tag_depth).uniq!
```
As uniq! returns any duplicate elements, or nil otherwise.

I'm a bit surprised that Array doesn't just provide `duplicates?` as it feels like a very common requirement.